python/fledge/plugins/north/azure_iot/azure_iot.py

```diff
@@ -89,74 +92,56 @@ def plugin_reconfigure():
 
 
 class AzureNorthPlugin(object):
-    """ North Azure Plugin """
+    """North Azure Plugin"""
 
-    def __init__(self):
+    def __init__(self, _config):
         self.event_loop = asyncio.get_event_loop()
+        self.config = _config
 
     async def send_payloads(self, payloads):
         is_data_sent = False
         last_object_id = 0
         num_sent = 0
-
-        MESSAGE_SIZE_LIMIT = 262144  # Limit form the Azure IoT Hub
-
+        message_size_limit = 262144  # Limit form the Azure IoT Hub
         size_payload_block = 0
-
         try:
-            payload_block = list()
-
+            payload_block = []
             for p in payloads:
-                read = dict()
-                read["asset"] = p['asset_code']
-                read["readings"] = p['reading']
-                read["timestamp"] = p['user_ts']
-                
+                read = {"asset": p['asset_code'], "readings": p['reading'], "timestamp": p['user_ts']}
                 size_payload_block += sys.getsizeof(json.dumps(read, separators=(',', ':')).encode('utf-8'))
-                if size_payload_block > MESSAGE_SIZE_LIMIT * 0.9: 
+                if size_payload_block > message_size_limit * 0.9:
                     # less than 90% of the maximum value is a quick solution to catch the not yet calculated
                     # overhead of the message class. Will be improved in a future version
                     _LOGGER.warning("The size of the message is larger than 256 kB! "
                                     "The remaining payloads will be sent on the next function call")
                     break
-                
                 last_object_id = p["id"]
                 payload_block.append(read)
-                
             num_sent = await self._send_payloads(payload_block)
             is_data_sent = True
         except Exception as ex:
-            _LOGGER.exception("Data could not be sent, %s", str(ex))
-
+            _LOGGER.exception("Data could not be sent, {}".format(str(ex)))
         return is_data_sent, last_object_id, num_sent
 
     async def _send_payloads(self, payload_block):
-        """ send a list of block payloads"""
-        
+        """send a list of block payloads"""
         num_count = 0
         try:
             device_client = IoTHubDeviceClient.create_from_connection_string(
-                config["primaryConnectionString"]["value"], websockets=config["websockets"]["value"])
-
+                self.config["primaryConnectionString"]["value"], websockets=self.config["websockets"]["value"])
             # Connect the device client.
             await device_client.connect()
-            
-            await self._send(device_client, payload_block)
+            message = Message(json.dumps(payload_block, separators=(',', ':')).encode('utf-8'))
```

Comment:
We should add an inline comment, why are we doing separators stuff

Reply:
As this plugin is contributed by ACDP. So not sure the exact reason though  comments added as per my understanding during the work.